src/events/SDL_mouse.c

```diff
@@ -739,18 +735,18 @@ static void SDL_PrivateSendMouseMotion(Uint64 timestamp, SDL_Window *window, SDL
                 y *= mouse->normal_speed_scale;
             }
         }
-        if (mouse->integer_mode >= 1) {
+        if (mouse->integer_mode_flags & 1) {
             // Accumulate the fractional relative motion and only process the integer portion
-            mouse->xrel_frac = SDL_modff(mouse->xrel_frac + x, &x);
-            mouse->yrel_frac = SDL_modff(mouse->yrel_frac + y, &y);
+            mouse->integer_mode_residual_motion[0] = SDL_modff(mouse->integer_mode_residual_motion[0] + x, &x);
```

Comment:
Please use named fields instead of an array of floats with unnamed indexes.

Reply:
If you really want to keep it as an array of floats, then please create named constants to use as indexes so it's clear what each entry is used for.